private/buf/bufwork/workspace_builder.go

```diff
@@ -64,9 +60,6 @@ func (w *workspaceBuilder) BuildWorkspace(
 	allModules := make([]bufmodule.Module, 0, len(workspaceConfig.Directories))
 	for _, directory := range workspaceConfig.Directories {
 		if cachedModule, ok := w.moduleCache[directory]; ok {
-			if directory == targetSubDirPath {
```

Comment:
For posterity: workspaces were being created that specifically *excluded* the "target" module - this caused other bugs down the line. I have no idea why this worked, but a workspace should include all the modules within it.